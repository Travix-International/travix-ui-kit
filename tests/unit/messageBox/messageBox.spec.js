import { shallow } from 'enzyme';
import React from 'react';
import MessageBox from '../../../components/messageBox/messageBox';

describe('MessageBox', () => {
  describe('#render()', () => {
    it('should render info message box with test modifications', () => {
      const mods = ['test'];
      const dataAttr = {
        gtmId: 'info-message',
        e2e: 'ui-messageBox',
      };
      const wrapper = shallow(
        <MessageBox dataAttr={dataAttr} mods={mods}>Just a simple message box</MessageBox>
      );

      expect(mods.length).toEqual(1);
      expect(mods[0]).toEqual('test');
      expect(wrapper).toMatchSnapshot();
    });

    it('should render success message box with title', () => {
      const wrapper = shallow(
        <MessageBox title="Success message box title" type="success">Success message box</MessageBox>
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('should render error message box with icon', () => {
      const wrapper = shallow(
        <MessageBox icon={<div className="icon" />} type="error">Error message box</MessageBox>
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('should render message box with provided className', () => {
      const wrapper = shallow(
        <MessageBox className={"test-cls"}>Message box</MessageBox>
      );

      expect(wrapper).toMatchSnapshot();
    });
  });
});
