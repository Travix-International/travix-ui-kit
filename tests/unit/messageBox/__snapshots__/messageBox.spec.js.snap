// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBox #render() should render error message box with icon 1`] = `
<div
  className="ui-messageBox ui-messageBox_error"
>
  <div
    className="ui-messageBox__content-icon"
  >
    <div
      className="icon"
    />
  </div>
  <div
    className="ui-messageBox__content"
  >
    <div
      className="ui-messageBox__content-body"
    >
      Error message box
    </div>
  </div>
</div>
`;

exports[`MessageBox #render() should render info message box with test modifications 1`] = `
<div
  className="ui-messageBox ui-messageBox_test ui-messageBox_info"
>
  <div
    className="ui-messageBox__content"
  >
    <div
      className="ui-messageBox__content-body"
    >
      Just a simple message box
    </div>
  </div>
</div>
`;

exports[`MessageBox #render() should render success message box with title 1`] = `
<div
  className="ui-messageBox ui-messageBox_success"
>
  <div
    className="ui-messageBox__content"
  >
    <div
      className="ui-messageBox__content-title"
    >
      Success message box title
    </div>
    <div
      className="ui-messageBox__content-body"
    >
      Success message box
    </div>
  </div>
</div>
`;

exports[`MessageBox #render() should render message box with provided className 1`] = `
<div
  className="ui-messageBox ui-messageBox_info test-cls"
>
  <div
    className="ui-messageBox__content"
  >
    <div
      className="ui-messageBox__content-body"
    >
      Message box
    </div>
  </div>
</div>
`;
